1.2.0

paulo-roger opened the pull request:
Everything working.

- Now using playlists for SFX
- The module detect a scene with old flags and inform the user.
- now supports weather-control in any language.

paulo-roger committed:
finished to implement playlists and patch to convert from the old sound configs to the new one.

paulo-roger committed:
playlists tested and working fine

paulo-roger committed:
testing with v9 and in a new world, fixed 1st bug

paulo-roger committed:
testing with v9 and in a new world, fixed 1st bug

paulo-roger committed:
testing with v9 and in a new world, fixed 1st bug

paulo-roger committed:
testing with v9 and in a new world, fixed 1st bug

paulo-roger committed:
testing with v9 and in a new world, fixed 1st bug

paulo-roger committed:
testing with v9 and in a new world, fixing bugs

paulo-roger committed:
testing with v9 and in a new world, fixing bugs

paulo-roger committed:
testing with v9 and in a new world, fixing bugs

paulo-roger committed:
unsetting unused flags

paulo-roger committed:
unsetting unused flags

paulo-roger committed:
unsetting unused flags

paulo-roger committed:
unsetting unused flags

paulo-roger committed:
unsetting unused flags

roi007leaf commented on scripts/patchPlaylist.js:
```diff
@@ -0,0 +1,49 @@
+export async function firstTime() {
+    Hooks.call(particleWeather, []);
+    FXMASTER.filters.setFilters([]);
+    if (canvas.scene.getFlag("weatherfx", "audio") !== undefined) {
```
not needed to check against undefined , can just be canvas.scene.getFlag("weatherfx", "audio")

paulo-roger replied:
you mean like:
```js
if (canvas.scene.getFlag("weatherfx", "audio")){
    ...
}
```
so if there is a value inside it consider the condition true?

paulo-roger replied:
just tested an example, gotcha! you're right.

roi007leaf commented on scripts/settings.js:
```diff
@@ -44,6 +49,62 @@ export function registerSettings() {
+        config: false,
+        type: String,
+        filePicker: 'audio',
+        default: 'modules/michaelghelfi/ambience/Snowing.ogg',
```
you can use blizzardSound variable instead of the string literal

paulo-roger replied:
Good, when I first wrote it I was in doubt, and forgot to test and change. thanks

roi007leaf commented on scripts/settings.js:
```diff
@@ -44,6 +49,62 @@ export function registerSettings() {
+        config: false,
+        type: String,
+        filePicker: 'audio',
+        default: 'modules/ivan-duch-music-packs/audio/rain-sfx.ogg',
```
same here

roi007leaf commented on scripts/settings.js:
```diff
@@ -44,6 +49,62 @@ export function registerSettings() {
+        config: false,
+        type: String,
+        filePicker: 'audio',
+        default: 'modules/michaelghelfi/ambience/RainandThunder.ogg',
```
same here

roi007leaf commented on scripts/settings.js:
```diff
@@ -44,6 +49,62 @@ export function registerSettings() {
+        config: false,
+        type: String,
+        filePicker: 'audio',
+        default: 'modules/ivan-duch-music-packs/audio/rain-sfx.ogg',
```
same here

roi007leaf commented on scripts/weatherfx.js:
```diff
@@ -29,16 +30,13 @@ Hooks.once('ready', async function () {
-        if (enableSound)
-            if (canvas.scene.getFlag("weatherfx", "audio") != "")
-                AudioHelper.play({ src: canvas.scene.getFlag("weatherfx", "audio"), volume: 0.8, loop: true }, true);
+    if (await canvas.scene.getFlag("weatherfx", "active") !== undefined || await canvas.scene.getFlag("weatherfx", "audio") !== undefined || await canvas.scene.getFlag("weatherfx", "currentWeather") !== undefined)
```
no need to check against undefined again

roi007leaf replied:
Do you check if the flag exists or doesnt exist? if it's the first time isnt it the case that the flag will not exist?

paulo-roger replied:
This is because the flag will exist only in scenes that used the older version of the module. so we get ride of those.

paulo-roger replied:
regarding the undefined condition did you tried to test it differently? when I did in other case it didn't go well, maybe it is foundry or maybe its me recalling wrong.

roi007leaf replied:
If get Flag returns a value it will have a value if not it will return undefined (assuming the get Flag function actually returns undefined)

paulo-roger replied:
I will test it to see how it goes and report back

paulo-roger replied:
OK only for this one though:
```js
canvas.scene.getFlag("weatherfx", "active")
```
I need to check if this is `!== undefined` because the value inside might be boolean `false`.

roi007leaf replied:
Boolean false is the same as the value returned from the function equals undefined

roi007leaf commented on scripts/weatherfx.js:
```diff
@@ -50,25 +48,23 @@ Hooks.once('renderWeatherApplication', async function (app, html, data) {
-Hooks.on('createChatMessage', async function (message, html, data) {
+Hooks.on('createChatMessage', async function (message) {
     let todaysWeather = await langJson()
     todaysWeather = todaysWeather["wctrl.weather.tracker.Today"]
```
better to move the string to a variable so it will be easier to maintain

paulo-roger replied:
thanks,  I will.

paulo-roger replied:
just to make sure before I change it: you mean the `message` string?

roi007leaf replied:
All the string wctrl....

roi007leaf commented on scripts/weatherfx.js:
```diff
@@ -251,9 +234,12 @@ async function checkWeather(msgString) {
-    clearEffects();
-    canvas.scene.setFlag("weatherfx", "active", true);
+async function weatherEffects(effectCondition) {
+    if (canvas.scene.getFlag("weatherfx", "isActive") !== undefined)
```
same here regarding undefined

paulo-roger replied:
Same as [here](https://github.com/LeafWulf/weatherfx/pull/1#discussion_r983997882) I am checking if the flag already exist so I know this scene had a weather effect applied at least once. Why though? 
I was getting an error when I run `function clearEffects()` because the effect condition flag would be undefined.
now it occurs to me though that if this is not a good practice and we have to remove it, we could change it to
 ```js
  if (canvas.scene.getFlag("weatherfx", "effectCondition")){
    ...
}
  ```

roi007leaf replied:
Yup

roi007leaf commented on scripts/weatherfx.js:
```diff
@@ -262,14 +248,10 @@ function weatherEffects(effectCondition) {
     }
 
-    if (enableSound) {
+    if (enableSound && effectCondition.hasSound) {
```
does effectCondition always exist? if not better to have a null check here for it

paulo-roger replied:
always exist yes

roi007leaf commented on scripts/weatherfx.js:
```diff
@@ -309,18 +291,28 @@ async function jsonItem() {
-async function weatherfxPlaylist(playlistName) {
+async function weatherfxPlaylistExists() {
+    let playlist = game.playlists?.contents.find((p) => p.name === playlistName);
+    let playlistExists = playlist ? true : false;
```
no need for this variable, can just use playlist variable

paulo-roger replied:
I was worried because maybe there is a small slightest chance of the user already have a playlist with the same name? lol

roi007leaf replied:
Then you need to make sure that your Playlist name is very unique

paulo-roger replied:
I am using 'Weather FX Playlist' because it is also easy to recognize for the user, I want the user to be able to change the sounds and volume, etc. I alert though to not change playlist name nor sounds name.

roi007leaf replied:
Then it's better to set the name for the Playlist in the settings and let the user decide the Playlist he wants to use

paulo-roger replied:
that's good indeed. The code creates the playlist once the game is ready, so we could throw a pop up with a text box for the user give a new name if the name is already used. (which I think it is very unlikely, but not impossible).
because next thing the code does is adding the songs to the recently created playlist.

paulo-roger committed:
Update patchPlaylist.js

removed the check against undefined.
```js
if (canvas.scene.getFlag("weatherfx", "audio") !== undefined)
```
changed to:
```js
if (canvas.scene.getFlag("weatherfx", "audio"))
```

paulo-roger committed:
Update settings.js

paulo-roger committed:
Update weatherfx.js

paulo-roger committed:
changes after review
